src/Native/System.Native/pal_io.cpp

```diff
@@ -443,3 +464,93 @@ int32_t MksTemps(char* pathTemplate, int32_t suffixLength)
 {
     return mkstemps(pathTemplate, suffixLength);
 }
+
+static
+void ConvertPalPollFdToNative(PollFd& pal, pollfd& native)
+{
+    native.fd = pal.FileDescriptor;
+    native.events = pal.EventsToPollFor;
+    native.revents = pal.ReceivedEvents;
+}
+
+static
+void ConvertNativePollfdToPal(pollfd& native, PollFd& pal)
+{
+    pal.FileDescriptor = native.fd;
+    pal.EventsToPollFor = native.events;
+    pal.ReceivedEvents = native.revents;
+}
+
+extern "C"
+int32_t Poll(PollFd* pollData, uint32_t numberOfPollFds, int32_t timeout)
+{
+    // Convert our standardized pollfd to the native one
+    pollfd* fds = new pollfd[numberOfPollFds];
```

Comment:
I'm not currently aware of any additional planned uses of poll (sockets is using epoll), and if additional needs do come in, I expect we'd either have a small number of fds (1 or 2) or we'd try to use epoll.  I think it'd make sense to avoid the allocation and have a small cap we trip over if we do need more dynamism, at which point we could add an implementation like Jon currently has.  My $.02.